airflow/migrations/versions/52d53670a240_fix_mssql_exec_date_rendered_task_instance.py

```diff
@@ -0,0 +1,74 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""fix_mssql_exec_date_rendered_task_instance_fields_for_MSSQL
+
+Revision ID: 52d53670a240
+Revises: 98271e7606e2
+Create Date: 2020-10-13 15:13:24.911486
+
+"""
+import sqlalchemy as sa
+from sqlalchemy.dialects import mssql
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = '52d53670a240'
+down_revision = '98271e7606e2'
+branch_labels = None
+depends_on = None
+
+TABLE_NAME = 'rendered_task_instance_fields'
+
+
+def upgrade():
+    """
+        Recreate RenderedTaskInstanceFields table changing timestamp to datetime2(6) when using MSSQL as backend
+    """
+    conn = op.get_bind()
+    if conn.dialect.name == "mssql":
+            json_type = sa.Text
+            op.drop_table(TABLE_NAME)  # pylint: disable=no-member
```

Comment:
Do we really have to drop the whole table, instead of just changing the column type?

Can you link (here, not in code) the difference between DATETIME2 and what ever `sa.TIMESTAMP` gives?

Reply:
We cannot alter column 'execution_date' because it is 'timestamp'. We're basically stuck with what we have. Bear in mind that the way this is at the moment (sa.TIMESTAMP) doesn't work with MSSQL

Reply:
Also, I should add that this is because SQL Server doesn't allow it. Below is the official explanation from Microsoft documentation - hope this helps.

https://docs.microsoft.com/en-us/sql/t-sql/statements/alter-table-transact-sql?view=sql-server-ver15:

_****ALTER COLUMN
Specifies that the named column is to be changed or altered.

The modified column can't be:

A column with a timestamp data type.****_

Reply:
Wat.

Cool, thanks